### PR TITLE
Feat/documentation: integrate protocol results into documentation

### PR DESCRIPTION
**Pull request by rainer-exxcellent**

### CODE REVIEW

File: documents/architecture-decisions.md [hunk 1]
```diff
@@ -612,17 +625,147 @@ available as a template. An API will list all available templates to the user.
...
+new "draft" version is initially created (Predicted new version = target version).
+Exception: Version 0, until the first time "published".
+
+##### Konfiguration
+
+There should be a configuration that defines which type of versioning should be
+taken for new documents.
+For existing documents the ype of versioning can NOT change during lifetime,
+i.e. once Semantic Versioning,
+always Semantic Versioning (or analog for Integer Versioning).
```
Comment (cloeser) [inline documents/architecture-decisions.md:696]: "semantic versioning" and "integer versioning" -> lower case

File: documents/architecture-decisions.md [hunk 2]
```diff
@@ -612,17 +625,147 @@ available as a template. An API will list all available templates to the user.
...
+
+The 'tracking/revision_history' should be maintained in the backend
+
+- Semantic Versioning
+
+  - Pre 1.0.0
+    - When saving (both new creation and "normal" saving) a summary and legacy
+      version can be specified in a  modal window.
+    - The backend creates a new revision history element for each change with the
+        `current date`, the `summary` and the `legacy version` of the user.
```
Comment (valentindey) [inline documents/architecture-decisions.md:716]: I don't completely get that sentence. Does it mean that the `date` is set to the current date (timestamp) and `summary` and `legacy_version` are given by the user?

File: documents/architecture-decisions.md [hunk 3]
```diff
@@ -612,17 +625,147 @@ available as a template. An API will list all available templates to the user.
...
+    - For status transitions, that also result in a change of the version
+      number (e.g. Review → Approved, Approved → Draft), the backend also
+      inserts a new element. A generic text is inserted as summary. e.g..:
```
Comment (valentindey) [inline documents/architecture-decisions.md:719]:
```suggestion
      inserts a new element. A generic text is inserted as summary. e.g.:
```

File: documents/architecture-decisions.md [hunk 4]
```diff
@@ -612,17 +625,147 @@ available as a template. An API will list all available templates to the user.
...
+
+- Semantic Versioning
+
+  - Pre 1.0.0
+    - When saving (both new creation and "normal" saving) a summary and legacy
+      version can be specified in a  modal window.
+    - The backend creates a new revision history element for each change with the
+        `current date`, the `summary` and the `legacy version` of the user.
+    - For status transitions, that also result in a change of the version
+      number (e.g. Review → Approved, Approved → Draft), the backend also
```
Comment (valentindey) [inline documents/architecture-decisions.md:718]:
```suggestion
      number (e.g. `Review` → `Approved`, `Approved` → `Draft`), the backend also
```

File: documents/architecture-decisions.md [hunk 5]
```diff
@@ -612,17 +625,147 @@ available as a template. An API will list all available templates to the user.
...
+      inserts a new element. A generic text is inserted as summary. e.g..:
+      "Status changed from Review to Approved".
+    - When the status changes to Published, all old Revision History elements
```
Comment (valentindey) [inline documents/architecture-decisions.md:721]:
```suggestion
    - When the status changes to `Published`, all old Revision History elements
```

File: documents/architecture-decisions.md [hunk 6]
```diff
@@ -612,17 +625,147 @@ available as a template. An API will list all available templates to the user.
...
+
+- Integer Versioning
+
+    Initially a revision history element is created. If an element with the
+    current version number already exists, none is created. New elements are
+    only created if the version number increases.
+    Behavior analogous to Semantic Versioning.
+
+We are in the Published status:
+When switching to Draft workflow status, a new Revision History Item is created.
```
Comment (valentindey) [inline documents/architecture-decisions.md:750]:
```suggestion
When in the `Published` status:
When switching to `Draft` workflow status, a new Revision History Item is created.
```

**Review by valentindey — CHANGES_REQUESTED** [review]

File: documents/architecture-decisions.md [hunk 7]
```diff
@@ -612,17 +628,149 @@ available as a template. An API will list all available templates to the user.
...
+The `tracking/revision_history` should be maintained in the backend
+
+- Semantic Versioning
+
+  - Pre 1.0.0
+    - When saving (both new creation and "normal" saving) a summary and legacy
+      version can be specified in a modal window.
+    - The backend creates a new revision history element for each change.
+      In this element the `date` is the current date, and the `summary`
+      and the `legacy version` are taken from of the modal window.
```
Comment (valentindey) [inline documents/architecture-decisions.md:721]:
```suggestion
      and the `legacy_version` are taken from of the modal window.
```

**Review by valentindey — APPROVED** [review]

**Review by valentindey — APPROVED** [review]

**Pull request merged**